### PR TITLE
chore: add note about React 18 automatic batching to useSubscription docs

### PR DESCRIPTION
**Pull request by alessbell**
Closes https://github.com/apollographql/apollo-client/issues/10001. See conversation on https://github.com/apollographql/apollo-client/pull/11066 for complete context.

Link to deployed preview: https://deploy-preview-11072--apollo-client-docs.netlify.app/api/react/hooks#subscriptions-and-react-18-automatic-batching

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Review by jerelmiller — APPROVED** [review]
This is a FANTASTIC addition to the subscription docs. Really well done. I had one minor thing to possibly add to make sure users of v3.6 and below can take advantage of this, but otherwise this looks great.

File: docs/source/api/react/hooks.mdx [hunk 1]
```diff
@@ -272,6 +272,61 @@ function DontReadTheComments({ repoFullName }) {
...
+
+Instead of using `useEffect` here, we can re-write this component to use the `onData` callback function accepted in `useSubscription`'s `options` object:
+
+```jsx
+export function Subscriptions() {
+  const [accumulatedData, setAccumulatedData] = useState([]);
+  const { data, error, loading } = useSubscription(
+    query,
+    {
+      onData({ data }) {
```
[the remark this answers is not in the document]
Comment (alessbell) [reply docs/source/api/react/hooks.mdx:313]: Yes good call! Adding a note about that now :)

**Comment by phryneas** [comment]
Love this solution :)

**Comment by c10b10** [comment]
Understood. I can confirm that using `onData` on the example repo will fix the issue. Thank you all!

**Pull request merged**